api/chat_main_handler.go

```diff
@@ -285,7 +262,7 @@ func (h *ChatHandler) OpenAIChatCompletionAPIWithStreamHandler(w http.ResponseWr
 	} else {
 
 		// Set up SSE headers
-		answerText, answerID, shouldReturn := chat_stream(ctx, chatSession, msgs, w)
+		answerText, answerID, shouldReturn := chat_stream(w, chatSession, msgs)
 		if shouldReturn {
 			return
 		}
```

Comment:
:robot: OpenAI

This patch changes the call to `chat_stream` to pass the response writer and chat session as arguments instead of using the context.

This change is an improvement because it makes the code more modular and easier to test.

Please confirm that the `chat_stream` function is defined and implemented correctly before merging this patch.

Also, please update the comments on the pull request to reflect this change.

<!-- This is an auto-generated comment by OpenAI -->